app/controllers/gobierto_admin/gobierto_plans/projects_controller.rb

```diff
@@ -0,0 +1,220 @@
+# frozen_string_literal: true
+
+module GobiertoAdmin
+  module GobiertoPlans
+    class ProjectsController < GobiertoAdmin::GobiertoPlans::BaseController
+      before_action :find_plan
+      before_action -> { module_allowed_action!(current_admin, current_admin_module, :edit) }, only: [:new, :create, :destroy]
+      before_action -> { module_allowed_action!(current_admin, current_admin_module, [:edit, :moderate]) }, only: [:index, :edit, :update]
+
+      helper_method :current_admin_actions
+
+      def index
+        find_plan
+        set_filters
+
+        @projects = @relation
+      end
+
+      def edit
+        find_plan
+        find_versioned_project
+        @project_visibility_levels = project_visibility_levels
+
+        @project_form = NodeForm.new(
+          @project.attributes.except(*ignored_project_attributes).merge(
+            plan_id: @plan.id,
+            options_json: @project.options,
+            admin: current_admin
+          )
+        )
+      end
+
+      def update
+        find_plan
+        @project = @plan.nodes.find params[:id]
+        @project_form = NodeForm.new(project_params.merge(id: params[:id], plan_id: params[:plan_id], admin: current_admin))
+
+        if @project_form.save
+          success_message = if suggest_unpublish?
+                              t(".suggest_unpublish_html", url: unpublish_admin_plans_plan_project_path(@plan, @project))
+                            else
+                              t(".success")
+                            end
+          redirect_to(
+            edit_admin_plans_plan_project_path(@plan, @project),
+            notice: success_message
+          )
+        else
+          render :edit
+        end
+      end
+
+      def unpublish
+        moderation_visibility_action(:draft)
+      end
+
+      def publish
+        moderation_visibility_action(:published)
+      end
+
+      def new
+        find_plan
+        @project_visibility_levels = project_visibility_levels
+
+        @project_form = NodeForm.new(
+          plan_id: @plan.id,
+          options_json: {},
+          admin: current_admin
+        )
+      end
+
+      def create
+        find_plan
+
+        @project_form = NodeForm.new(project_params.merge(id: params[:id], plan_id: params[:plan_id], admin: current_admin))
+
+        if @project_form.save
+          redirect_to(
+            edit_admin_plans_plan_project_path(@plan, @project_form.node),
+            notice: t(".success")
+          )
+        else
+          @project_visibility_levels = project_visibility_levels
+
+          render :new
+        end
+      end
+
+      def destroy
+        find_plan
+        set_filters
+
+        @project = @plan.nodes.find params[:id]
+
+        raise_action_not_allowed unless current_admin_actions.include? :destroy
```

Comment:
You could move this line to the first line in the method to avoid unnecessary queries.

Reply:
Or to a a before_action